### PR TITLE
Make dry_run return transparent Receipt

### PR DESCRIPTION
**Pull request by Salka1988**
blocked by https://github.com/FuelLabs/fuel-vm/pull/233

### CODE REVIEW

File: fuel-client/src/client.rs [hunk 1]
```diff
@@ -156,7 +156,9 @@ impl FuelClient {
 
     /// Default dry run, matching the exact configuration as the node
     pub async fn dry_run(&self, tx: &Transaction) -> io::Result<Vec<Receipt>> {
-        self.dry_run_opt(tx, None).await
+        let a = self.dry_run_opt(tx, None).await;
+        dbg!(&a);
```
Comment (leviathanbeak) [inline fuel-client/src/client.rs:160]: no-no

File: fuel-client/src/client/schema/snapshots/fuel_gql_client__client__schema__tx__tests__dry_run_tx_gql_output.snap [hunk 2]
```diff
@@ -1,12 +0,0 @@
----
-source: fuel-client/src/client/schema/tx.rs
```
Comment (leviathanbeak) [inline fuel-client/src/client/schema/snapshots/fuel_gql_client__client__schema__tx__tests__dry_run_tx_gql_output.snap:2]: shouldn't new snapshot tests be generated ?

Comment (Salka1988) [reply fuel-client/src/client/schema/snapshots/fuel_gql_client__client__schema__tx__tests__dry_run_tx_gql_output.snap:2]: I replaced the new one locally with the old one, so I smoked it. The one here is actually new. I didn't know that I could do that with `cargo insta`.

File: fuel-client/src/client.rs (hunk 1 above)
Comment (Salka1988) [reply fuel-client/src/client.rs]: deleted! :)

File: fuel-tests/tests/tx/utxo_validation.rs [hunk 3]
```diff
@@ -100,9 +100,10 @@ async fn submit_utxo_verified_tx_with_min_gas_price() {
         }
 
         // Once https://github.com/FuelLabs/fuel-core/issues/50 is resolved this should rely on the Submitted Status rather than Success
+
         assert!(matches!(
             transaction_result,
-            TransactionStatus::Success { .. }
```
Comment (xgreenx) [inline fuel-tests/tests/tx/utxo_validation.rs:105]: The test should be positive, you need to increase the gas limit

File: fuel-tests/tests/tx/utxo_validation.rs [hunk 4]
```diff
@@ -172,9 +173,9 @@ async fn dry_run_override_utxo_validation() {
     assert_eq!(2, log.len());
 
     assert!(matches!(log[0],
-        Receipt::Return {
-            val, ..
-        } if val == 1));
+        Receipt::Panic {
```
Comment (xgreenx) [inline fuel-tests/tests/tx/utxo_validation.rs:176]: The test should be positive, you need to increase the gas limit

Comment (Salka1988) [reply fuel-tests/tests/tx/utxo_validation.rs]: fixed!

File: fuel-tests/tests/tx/utxo_validation.rs (hunk 3 above)
Comment (Salka1988) [reply fuel-tests/tests/tx/utxo_validation.rs]: fixed!

File: fuel-client/src/client/schema/tx.rs [hunk 5]
```diff
@@ -260,7 +262,7 @@ pub struct DryRunArg {
 )]
 pub struct DryRun {
     #[arguments(tx = &args.tx, utxo_validation = &args.utxo_validation)]
-    pub dry_run: Vec<OpaqueReceipt>,
+    pub dry_run: Vec<transparent_receipt::Receipt>,
```
Comment (xgreenx) [inline fuel-client/src/client/schema/tx.rs:265]: Why did you change the return type? (Sorry, I don't familiar with this code, so I wonder why we need changes here)

File: Cargo.lock [hunk 6]
```diff
@@ -113,7 +113,7 @@ version = "0.7.6"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "fcb51a0695d8f838b1ee009b3fbf66bda078cd64590202a864a8f3e8c4315c47"
 dependencies = [
- "getrandom 0.2.7",
+ "getrandom 0.2.8",
```
Comment (xgreenx) [inline Cargo.lock:116]: Could you revert those bumping of all deps, please? Better to do that in a separate PR to not mix things. Or do you need some new versions of deps?

File: fuel-client/src/client/schema/tx.rs (hunk 5 above)
Comment (Salka1988) [reply fuel-client/src/client/schema/tx.rs:265]: https://github.com/FuelLabs/fuel-tx/pull/198#issuecomment-1273720252

File: Cargo.lock (hunk 6 above)
Comment (Salka1988) [reply Cargo.lock:116]: I didn't change this. I think this goes automatically :thinking:

Comment (xgreenx) [reply Cargo.lock:116]: You can revert this file then=)

File: fuel-core/src/schema/tx/receipt.rs [hunk 7]
```diff
@@ -141,6 +142,9 @@ impl Receipt {
     async fn nonce(&self) -> Option<Bytes32> {
         self.0.nonce().copied().map(Bytes32)
     }
+    async fn contract_id(&self) -> Option<ContractId> {
```
Comment (xgreenx) [inline fuel-core/src/schema/tx/receipt.rs:145]:
```suggestion
    }

    async fn contract_id(&self) -> Option<ContractId> {
```

File: Cargo.toml [hunk 8]
```diff
@@ -21,6 +21,9 @@ members = [
   "xtask",
 ]
 
+[patch.crates-io]
```
Comment (xgreenx) [inline Cargo.toml:24]: Before merge we need to remove this=) Created PR with release https://github.com/FuelLabs/fuel-vm/pull/239

File: fuel-client/src/client/schema/snapshots/fuel_gql_client__client__schema__tx__tests__transparent_transaction_by_id_query_gql_output.snap [hunk 9]
```diff
@@ -4,83 +4,10 @@ expression: operation.query
...
-        amount
-        assetId
-      }
-      ... on ContractCreated {
-        contract {
-          id
-        }
-        stateRoot
-      }
+    rawPayload
```
Comment (xgreenx) [inline fuel-client/src/client/schema/snapshots/fuel_gql_client__client__schema__tx__tests__transparent_transaction_by_id_query_gql_output.snap:7]: Hmm, I'm not sure why do we have changes here=D

Comment (Salka1988) [reply fuel-client/src/client/schema/snapshots/fuel_gql_client__client__schema__tx__tests__transparent_transaction_by_id_query_gql_output.snap:7]: Freshly generated .snap file based on the new settings. I don't know why he generated it like this, but I could theorize :D

File: fuel-client/src/client/schema/tx.rs [hunk 10]
```diff
@@ -280,13 +282,10 @@ pub mod tests {
...
     use fuel_vm::fuel_types::bytes::SerializableVec;
 
-    pub mod transparent_receipt;
-    pub mod transparent_tx;
-
     #[test]
     fn transparent_transaction_by_id_query_gql_output() {
         use cynic::QueryBuilder;
-        let operation = transparent_tx::TransactionQuery::build(TxIdArgs {
+        let operation = TransactionQuery::build(TxIdArgs {
```
Comment (xgreenx) [inline fuel-client/src/client/schema/tx.rs:288]: Okay, I found! It is because you changed the type here=) And question why?

File: Cargo.toml (hunk 8 above)
Comment (xgreenx) [reply Cargo.toml:24]: You can use new release now=) https://github.com/FuelLabs/fuel-vm/releases/tag/v0.19.0

File: fuel-client/src/client/schema/snapshots/fuel_gql_client__client__schema__tx__tests__transparent_transaction_by_id_query_gql_output.snap (hunk 9 above)
Comment (Voxelot) [reply fuel-client/src/client/schema/snapshots/fuel_gql_client__client__schema__tx__tests__transparent_transaction_by_id_query_gql_output.snap:7]: The transparent types mean we use the typed fields rather than canonical "rawPayload" bytes. By moving this query to use the "rawPayload", it is no longer transparent and we lose our test coverage of these fields on the graphql api.

File: fuel-client/src/client/schema/tx.rs (hunk 10 above)
Comment (Salka1988) [reply fuel-client/src/client/schema/tx.rs]: Reverted! :)

**Review by xgreenx — APPROVED** [review]

**Review by leviathanbeak — APPROVED** [review]
nice one Mr Salka

**Review by Voxelot — APPROVED** [review]
